viamrobotics/viam-cartographer
# RSDK-3276/RSDK-3277: Update cartographer for data replay

### Summary
This PR covers the following two tickets:

1. [RSDK-3276](https://viam.atlassian.net/browse/RSDK-3276): viam-cartographer should read from sensor even when use_live_data = false, and not wait 200 ms
2. [RSDK-3277](https://viam.atlassian.net/browse/RSDK-3277): Cartographer should keep polling for new data even when use_live_data=false

It also modifies an existing test for offline mode that was broken by this change.

### Manual Testing
I tested with a dataset of about ~1000 images and confirmed all of the images were downloaded (by looking at the data directory) and fed into cartographer (by printing this out - technically all but the last one are fed in, which is expected given my implementation). I confirmed that the time taken to build the map is similar to offline mode (a couple seconds) and that the map is built correctly.

[RSDK-3276]: https://viam.atlassian.net/browse/RSDK-3276?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RSDK-3277]: https://viam.atlassian.net/browse/RSDK-3277?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
